File: Sorting/selectionSort.cpp
#include<iostream>
#include<vector>
using namespace std;

int getMinimumIndex(vector<int> &arr, int i){
	int minIndex = i;
	for(int j=i+1; j<arr.size(); j++){
		if(arr[j]<arr[minIndex]) minIndex = j;
	}
	return minIndex;
}

void selectionSort(vector<int> &arr){
	for(int i=0; i<arr.size(); i++){
		int minIndex = getMinimumIndex(arr, i);
		if(i!=minIndex) swap(arr[i], arr[minIndex]);
	}
}

int main(){
	vector<int> arr = {-2, 8, 0, 3, 2, 81, 0};
	selectionSort(arr);
	for(auto e:arr) cout<<e<<" ";
	cout<<endl;
}

File: Sorting/selectionSort.js
function getMinIndex(arr, i){
	let minIndex = i;
	for(let j=i+1; j<arr.length; j++){
		if(arr[j]<arr[minIndex]) minIndex = j;
	}
	return minIndex;
}
function selectionSort(arr){
	for(let i=0; i<arr.length; i++){
		let minIndex = getMinIndex(arr, i);
		if(i!==minIndex){
			let temp = arr[i];
			arr[i] = arr[minIndex];
			arr[minIndex] = temp;
		}
	}
}

let arr = [99, 23, -1, 0, 13, 88, -4, 6, 8];
selectionSort(arr);
console.log(arr);